src/parser.rs

```diff
@@ -94,6 +115,15 @@ pub fn variable(input: &str) -> IResult<&str, Factor, Error<&str>> {
     map(alpha1, |name: &str| Factor::Variable(name.to_string()))(input)
 }
 
+// variables should not be called "true" or "false" because they are KW
+pub fn variable2(input: &str) -> IResult<&str, Factor, Error<&str>> {
```

Comment:
Use a separate parser to identify all the languages' keywords, and rewrite the 'variable' parser to just accept every `alpha1` except `keywords` parser